stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -17,26 +18,23 @@ class StockReleaseChannel(models.Model):
     )
 
     def filter_release_channel_partner_window(self, picking, partner):
-        channels = self
-        if (
-            not partner.delivery_time_preference
-            or partner.delivery_time_preference == "anytime"
-        ):
-            return channels
-
-        for channel in self:
-            if not channel.shipment_date:
-                continue
-            shipment_datetime = fields.Datetime.to_datetime(channel.shipment_date)
-            if channel.process_end_date:
-                shipment_datetime = shipment_datetime.replace(
-                    hour=channel.process_end_date.hour,
-                    minute=channel.process_end_date.minute,
+        if picking.sale_id.commitment_date:
+            # Date is forced on SO, don't filter
+            return self
+        if partner.delivery_time_preference == "anytime":
+            return self
+        if partner.delivery_time_preference == "workdays":
+            open_days = set(range(5))
+        else:
+            open_days = set(
+                map(
+                    int,
+                    partner.delivery_time_window_ids.time_window_weekday_ids.mapped(
+                        "name"
+                    ),
                 )
-            if (
-                channel.respect_partner_delivery_time_windows
-                and not picking.sale_id.commitment_date
-                and not partner.is_in_delivery_window(shipment_datetime)
-            ):
-                channels -= channel
-        return channels
+            )
+        return self.filtered(
+            lambda channel: not channel.respect_partner_delivery_time_windows
+            or (channel.shipment_date and channel.shipment_date.weekday() in open_days)
```

Comment:
Are we sure about this? I mean… It looks like to me that shipment date is compared with partner's open days.
What if shipped on friday when delivery lead is 1 day?

Reply:
then the naming is probably wrong, right? I would rename it reception_date in a future version, this is confusing.